### PR TITLE
Convert Excel sheet raw data into processed JSON

### PR DESCRIPTION
**Pull request by gotlougit**
This relies on PR #3's output, which is an Excel sheet full of the Google Drive links of all the material inside the given link. The code classifies material based on parent folder and MIME type, so if it's inside ```Reference Books/``` folder, it's probably a book, or if it's a PPTX file, it's probably a lecture presentation. If we get more parents from the output of PR #3, the code will still be able to work just fine. Some more cleanup may be required though.

### CODE REVIEW

File: src/parseexcel.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+pptmimeList = ["application/vnd.ms-powerpoint","application/vnd.openxmlformats-officedocument.presentationml.presentation"]
+
+outfile = open("links.json","w")
+
+for i in range(len(x.index)):
+    
+    ID = x.loc[i]["id"]
```
Comment (maan2003) [inline src/parseexcel.py]: nit: use snake case for variables
```suggestion
    id = x.loc[i]["id"]
```

File: src/parseexcel.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    tag = json.loads(x.loc[i]["tags"].replace("'",'"'))
+
+    newEntry = [ID, name, mime, tag]
+    newtag = []
+
+    typicalAncestors = []
+    for c in coursecodes:
+        for t in tag:
+            if c in t:
+                typicalAncestors += [t]
```
Comment (maan2003) [inline src/parseexcel.py]: append?
```suggestion
                typicalAncestors.append(t)
```

File: src/parseexcel.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    if mime in pptmimeList and "kind:lecture-ppt" not in newtag:
+        newtag.append("kind:lecture-ppt")
+    
+    for t in tag:
+        if t not in typicalAncestors and t:
+            newtag.append("other:"+t)
+
+    newEntry[-1] = newtag
+    outfile.write(json.dumps(newEntry))
+    outfile.write("\n")
```
Comment (maan2003) [inline src/parseexcel.py]: lets use a big json array, instead of one json per line.

File: src/parseexcel.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
... 'MEPE11', 'MEPE12', 'MEPE13', 'MEPE14', 'MEPE15', 'MEPC22', 'MEPC23', 'MEPC24', 'MEPC25', 'MELR18', 'MELR19', 'MELR20', 'MEPE16', 'MEPE17', 'MEPE18', 'MEPE19', 'MEPE20', 'MEPE21', 'MEPC26', 'MEPC27', 'MEPC28', 'MELR21', 'MEPE22', 'MEPE23', 'MEPE25', 'MEPE26', 'MEPE27', 'MEPE', 'MEPC29', 'MEPE29', 'MEPE30', 'MEPE31', 'MEPE32', 'MEPE33', 'MEPE37', 'MEPE39', 'MEPE41', 'MELR14', 'PHIR11', 'PHIR12A', 'PHIR12B', 'PHIR12C', 'PHIR12', 'PHOE10', 'PHOE11', 'PHOE', 'PHOE13', 'PHOE14', 'PHOE15', 'PHOE16', 'PRIR11', 'PRPC10', 'PRPC11', 'PRPC12', 'PRPC13', 'PRPC14', 'PRPC15', 'PRPC16', 'PRPC17', 'PRPC19', 'PRPL10', 'PRPL11', 'PRPL12', 'PRPL13', 'PRPC18', 'PRPC20', 'PRPC21', 'PRPL14', 'PRPL15', 'PRPL16', 'PRPE10', 'PRPE11', 'PRPE12', 'PRPE13', 'PRPE14', 'PRPE15', 'PRPC22', 'PRPC23', 'PRPC24', 'PRPC25', 'PRPL17', 'PRPL18', 'PRPE16', 'PRPE17', 'PRPE18', 'PRPE19', 'PRPE20', 'PRPE21', 'PRPC26', 'PRPC27', 'PRPC28', 'PRPL', 'PRPL20', 'PRPE23', 'PRPE24', 'PRPE', 'PRPE27', 'PRPE28', 'PRPE29', 'PRPE30', 'PRPC29', 'PRPE33', 'PRPE34', 'PRPE35', 'PRPE37', 'PRPE39', 'PRPE40', 'PRPE41', 'PRPE43', 'PRPE45', 'HSIR11', 'HSIR12', 'PHIR14', 'PHIR13', 'MAIR12','MAIR11', 'MAIR22','MAIR24','CEPC23','MAIR21','ECIR11']
```
Comment (maan2003) [inline src/parseexcel.py]: put this in a separate file. spare people with line wrapping on!

**Review by maan2003 — APPROVED** [review]
LGTM! just some minor nits.

File: src/parseexcel.py (hunk 4 above)
Comment (gotlougit) [reply src/parseexcel.py]: Got it, latest commit has all these changes now

**Pull request merged**